Docs/source/building/spack.rst

```diff
@@ -1,40 +1,49 @@
 Building WarpX with Spack
 ===============================
 
-WarpX can be installed using Spack. From the Spack web page: "Spack is a package management tool designed to support multiple
-versions and configurations of software on a wide variety of platforms and environments."
+WarpX can be installed using Spack.
+From the `Spack web page <https://spack.io>`_: "Spack is a package management tool designed to support multiple versions and configurations of software on a wide variety of platforms and environments."
 
-Spack is available from `github <https://github.com/spack/spack>`__. Spack only needs to be cloned and can be used right away - there are no installation
-steps. The spack command, "spack/bin/spack", can be used directly or "spack/bin" can be added to your execute path.
+.. note::
+
+   Quick-start hint for macOS users:
+   Before getting started with Spack, please check what you manually installed in ``/usr/local``.
+   If you find entries in ``bin/``, ``lib/`` et al. that look like you manually installed MPI, HDF5 or other software at some point, then remove those files first.
```

Comment:
```suggestion
   If you find entries in ``bin/``, ``lib/`` et al. that look like you manually installed MPI, HDF5 or other software at some point, then consider removing those files first, for a cleaner installation.
```

I made this change because, for example, I needed to use the `libiconv` from /usr/lib in my all-space installation to get python installed.
I did it by by adding to  ~/.spack/darwin/packages.yaml:
```vim
packages:
  libiconv:
    buildable: False
    paths:
      libiconv: /usr
```

Reply:
Interesting work-around, thanks for sharin. Sounds like a recipe bug to me... Are you building a specific version of python or the current default (`3.7.6`)?

Reply:
It is the current default, built with spack without any other special instruction.

Reply:
And you build with clang or GCC from homebrew (which version)? Can you please post your full `packages.yaml`?
I am trying to reproduce this in https://github.com/ax3l/spack-emacs/runs/603445037?check_suite_focus=true

Re the wording changes: not removing/cleaning up did prevent on two people's macOS last week (no names ;-) ) to successfully compile, so I would not word it "too optional" like "can be done later" :)

Reply:
With clang, I am using:
```bash
$ cat ~/.spack/darwin/packages.yaml
packages:
  libiconv:
    buildable: False
    paths:
      libiconv: /usr
  all:
    providers:
      mpi: [mpich, openmpi]
```
Good point!

Reply:
Thanks! Fixed in https://github.com/spack/spack/pull/16193 :)